src/test/clojure/clara/test_accumulators.clj

```diff
@@ -59,10 +59,15 @@
         session (-> empty-session
                     (insert (->Temperature 30 "MCI"))
                     (insert (->Temperature 10 "MCI"))
-                    (insert (->Temperature 80 "MCI")))
+                    (insert (->Temperature 80 "MCI"))
+                    fire-rules)
```

Comment:
I agree, the behavior has never been reliable.  I don't like adding more options and tricky semantics for things like this either really.  If it is any consolation, prior to Drools 6, the entire engine was very eager in computing LHS logic and accumulators.  When Drools 6 released, it went nearly the complete opposite direction and it was necessary to fire rules to observe almost any behavior.

Drools has a large user-base too.

I don't like making "breaking" changes, but this is really undocumented and unreliable behavior prior to these changes.  If anything, the documentation should make it clearly stated that prior to firing rules, the state of working memory is undefined.  The engine reserves this as an implementation detail to facilitate better runtime optimizations.